examples/restore_login.py

```diff
@@ -1,71 +1,90 @@
 
-def write_details_to_disk(resp: LoginResponse) -> None:
+
+def write_details_to_disk(resp: LoginResponse, home_server) -> None:
     """Writes the required login details to disk so we can log in later without
     using a password.
 
     Arguments:
         resp {LoginResponse} -- the successful client login response.
+        home_server -- the name of the homeserver, e.g. "https://matrix.example.org"
     """
     # open the config file in write-mode
     with open(CONFIG_FILE, "w") as f:
         # write the login details to disk
         json.dump(
             {
-            "access_token": resp.access_token,
-            "device_id": resp.device_id,
-            "user_id": resp.user_id
+                "home_server": home_server,  # e.g. "https://matrix.example.org"
+                "access_token": resp.access_token,  # long cryptographic access token
+                "device_id": resp.device_id,  # device ID, 10 uppercase letters
+                "user_id": resp.user_id  # e.g. "@user:example.org"
             },
             f
         )
 
 
 async def main() -> None:
-    client = AsyncClient("https://matrix.example.org", "@alice:example.org")
 
     # If there are no previously-saved credentials, we'll use the password
     if not os.path.exists(CONFIG_FILE):
-        resp = await client.login("hunter2")
+        print("First time use. Did not find credential file. Asking for homeserver, user, and password to create credential file.")
+        home_server = "https://matrix.example.org"
+        home_server = input(f"Enter your homeserver: [{home_server}] ")
+        if not (home_server.startswith("https://")
+                or home_server.startswith("https://")):
+            home_server = "https://" + home_server
+        user_id = "@user:example.org"
+        user_id = input(f"Enter your full user ID: [{user_id}] ")
+        client = AsyncClient(home_server, user_id)
+        pw = getpass.getpass()
+        resp = await client.login(pw)
         # check that we logged in succesfully
         if (isinstance(resp, LoginResponse)):
-            write_details_to_disk(resp)
+            write_details_to_disk(resp, home_server)
         else:
+            print(f"homeserver = \"{home_server}\"; user = \"{user_id}\"")
             print(f"Failed to log in: {resp}")
             sys.exit(1)
 
         print(
-            "Logged in using a password.",
-            "Try running the script again to login with an access token"
+            "Logged in using a password. Credentials were stored.",
+            "Try running the script again to login with credentials (an access token)."
         )
 
     # Otherwise the config file exists, so we'll use the stored credentials
     else:
         # open the file in read-only mode
         with open(CONFIG_FILE, "r") as f:
             config = json.load(f)
+            client = AsyncClient(config['home_server'])
             client.access_token = config['access_token']
```

Comment:
Let's use the [restore_login()](https://matrix-nio.readthedocs.io/en/latest/nio.html#nio.Client.restore_login) method here.

Reply:
I tried that, but I was not successful. 
I stumbled over error: 
nio.exceptions.LocalProtocolError: No store class was provided in the config.

I added ClientConfig module, lines like
ClientConfig(store_sync_tokens=True)
added store_path="."
etc
But to no avail. It seems that `restore_login()` requires a store_class to be set.

How to proceed? Any hints?
Since I failed, I left it as is for the time being.

Reply:
Does this have anything to do with the fact that I do not have libolm or e2ee installed?

Reply:
It has, but i think that's a mistake by me as well. It should work without a store class as well. So let's leave it as is for now.

Reply:
Suggested next step? Do you want me to make any other changes or ready for Pull?

Reply:
I'll fix nio so it works, but we'll leave the example as is for now.

Reply:
OK. Once you think we should give `restore_login()` another try, let me know. I can give it another shot then.